tooling/autodispose-error-prone-checker/src/main/java/com/uber/autodispose/error/prone/checker/MissingAutoDisposeErrorChecker.java

```diff
@@ -0,0 +1,175 @@
+        .add(instanceMethod().onDescendantOf("io.reactivex.Flowable").named(AS))
+        .add(instanceMethod().onDescendantOf("io.reactivex.Maybe").named(AS))
+        .build();
+
+    SUBSCRIBE_MATCHERS = new ImmutableList.Builder<MethodMatchers.MethodNameMatcher>()
+        .add(instanceMethod().onDescendantOf("io.reactivex.Single").named(SUBSCRIBE))
+        .add(instanceMethod().onDescendantOf("io.reactivex.Observable").named(SUBSCRIBE))
+        .add(instanceMethod().onDescendantOf("io.reactivex.Completable").named(SUBSCRIBE))
+        .add(instanceMethod().onDescendantOf("io.reactivex.Flowable").named(SUBSCRIBE))
+        .add(instanceMethod().onDescendantOf("io.reactivex.Maybe").named(SUBSCRIBE))
+        .build();
+  }
+
+  /**
+   * Matcher to find the as operator in the observable chain.
+   */
+  private static final Matcher<ExpressionTree> METHOD_NAME_MATCHERS =
+      new Matcher<ExpressionTree>() {
+        @Override
+        public boolean matches(ExpressionTree tree, VisitorState state) {
+          if (!(tree instanceof MethodInvocationTree)) {
+            return false;
+          }
+          MethodInvocationTree invTree = (MethodInvocationTree) tree;
+
+          final MemberSelectTree memberTree = (MemberSelectTree) invTree.getMethodSelect();
+          if (!memberTree.getIdentifier().contentEquals(AS)) {
+            return false;
+          }
+
+          for (MethodMatchers.MethodNameMatcher nameMatcher : AS_CALL_MATCHERS) {
+            if (nameMatcher.matches(invTree, state)) {
+              ExpressionTree arg = invTree.getArguments().get(0);
+              final Type scoper = state
+                  .getTypeFromString("com.uber.autodispose.AutoDisposeConverter");
+              return ASTHelpers.isSubtype(ASTHelpers.getType(arg), scoper, state);
+            }
+          }
+          return false;
+        }
+      };
+
+  private static Matcher<MethodInvocationTree> matcher(List<String> classesWithLifecycle) {
+    return new Matcher<MethodInvocationTree>() {
+      @Override
+      public boolean matches(MethodInvocationTree tree, VisitorState state) {
+
+        boolean matchFound = false;
+        final MemberSelectTree memberTree = (MemberSelectTree) tree.getMethodSelect();
+        if (!memberTree.getIdentifier().contentEquals(SUBSCRIBE)) {
+          return false;
+        }
+
+        for (MethodMatchers.MethodNameMatcher nameMatcher : SUBSCRIBE_MATCHERS) {
+          if (!nameMatcher.matches(tree, state)) {
+            continue;
+          } else {
+            matchFound = true;
+            break;
+          }
+        }
+        if (!matchFound) {
+          return false;
+        }
+
+        ClassTree enclosingClass =
+            ASTHelpers.findEnclosingNode(state.getPath(), ClassTree.class);
+        Type.ClassType enclosingClassType = ASTHelpers.getType(enclosingClass);
+
+        for (String s : classesWithLifecycle) {
```

Comment:
We can use a java 8 stream here

Reply:
https://github.com/uber/AutoDispose/pull/156/commits/702b9291de122752de71964e9d3311ca0f64b3b6